src/pymor/algorithms/symplectic.py

```diff
@@ -0,0 +1,514 @@
+
+    def __getitem__(self, ind):
+        assert self.E.check_ind(ind)
+        # check_symplecticity = False, otherwise recursion loop
+        return type(self)(self.E[ind], self.F[ind], check_symplecticity=False)
+
+    def lincomb(self, coefficients):
+        assert isinstance(coefficients, np.ndarray)
+        if coefficients.ndim == 1:
+            coefficients = coefficients[np.newaxis, ...]
+        assert len(coefficients.shape) == 2 and coefficients.shape[1] == len(self)
+        result = self.E.lincomb(coefficients[:, :len(self.E)])
+        result += self.F.lincomb(coefficients[:, len(self.E):])
+        return result
+
+    def extend(self, U, method='svd_like', modes=2, product=None):
+        """Extend the |SymplecticBasis| with vectors from a |VectorArray|.
+
+        Parameters
+        ----------
+        U
+            The vectors used for the extension as |VectorArray|.
+        method
+            The method used for extension. Available options are
+                ('svd_like', 'complex_svd', 'symplectic_gram_schmidt').
+        modes
+            Number of modes to extract from U. Has to be even.
+        product
+            A product to use for the projection error. Default is None.
+        """
+        from pymor.algorithms.symplectic import (psd_complex_svd,
+                                                 psd_svd_like_decomp,
+                                                 symplectic_gram_schmidt)
+        assert modes % 2 == 0, 'number of modes has to be even'
+        assert method in ('svd_like', 'complex_svd', 'symplectic_gram_schmidt')
+
+        U_proj_err = U - self.lincomb(U.inner(self.transposed_symplectic_inverse().to_array()))
+        proj_error = U_proj_err.norm(product=product)
+
+        if method in ('svd_like', 'complex_svd'):
+
+            if method == 'svd_like':
+                new_basis = psd_svd_like_decomp(U_proj_err, modes)
+            elif method == 'complex_svd':
+                new_basis = psd_complex_svd(U_proj_err, modes)
+
+            self.append(new_basis)
+
+        elif method == 'symplectic_gram_schmidt':
+            J = CanonicalSymplecticFormOperator(self.phase_space)
+            basis_length = len(self)
+            # find max error
+            idx = proj_error.argsort()[-modes//2:][::-1]
+            new_basis = U[idx].copy()
+            new_basis.scal(1/new_basis.norm())
+            new_basis.append(J.apply_adjoint(new_basis))
+            self.append(
+                SymplecticBasis.from_array(new_basis, check_symplecticity=False),
+                check_symplecticity=False,
+            )
+            symplectic_gram_schmidt(self.E, self.F, offset=basis_length, copy=False)
+        else:
+            assert False
+
+
+def psd_svd_like_decomp(U, modes, balance=True):
+    """Generates a |SymplecticBasis| with the PSD SVD-like decompostion.
+
+    This is an implementation of Algorithm 1 in :cite:`BBH19`.
```

Comment:
```suggestion
    This is an implementation of Algorithm 1 in :cite:`BBH2019`.
```

Reply:
Thanks for hinting in the right direction. Instead I renamed the bib entry since BBH19 seems to match the style of the other entries more than BBH2019. I built the docs locally and everything worked well. Please let me know if there are other issues since I could not access the link above due to access restrictions.

Reply:
Great, thanks. [This job](https://zivgitlab.uni-muenster.de/pymor/pymor/-/jobs/951083) should complete without issue then.
I just tried that link above to a job log and also got access denied. So it's not just you, maybe the job artefact got pruned.